contracts/TownHallZap.sol

```diff
@@ -28,4 +36,41 @@ contract TownHallZap {
             townHall.mint(to);
         }
     }
+
+    // exactInputSingle for ERC20
+    function convertAndMint(address sourceToken, address mintTo, uint256 amountInMaximum) external {
+        TransferHelper.safeTransferFrom(sourceToken, msg.sender, address(this), amountInMaximum);
+        TransferHelper.safeApprove(sourceToken, address(uniswapV3Router), amountInMaximum);
+
+        ISwapRouter.ExactOutputSingleParams memory params = ISwapRouter.ExactOutputSingleParams({
+            tokenIn: sourceToken,
+            tokenOut: address(huntToken),
+            fee: 3000,
+            recipient: msg.sender,
+            deadline: block.timestamp,
+            amountOut: LOCK_UP_AMOUNT,
+            amountInMaximum: amountInMaximum,
+            sqrtPriceLimitX96: 0
+        });
+
+        uniswapV3Router.exactOutputSingle(params);
+        townHall.mint(mintTo);
+    }
+
+    // exactInputSingle for ETH
+    function convertAndMintETH(address mintTo, uint256 amountInMaximum) external payable {
+        ISwapRouter.ExactOutputSingleParams memory params = ISwapRouter.ExactOutputSingleParams({
+            tokenIn: WETH,
+            tokenOut: address(huntToken),
+            fee: 3000,
+            recipient: msg.sender,
+            deadline: block.timestamp,
+            amountOut: LOCK_UP_AMOUNT,
+            amountInMaximum: amountInMaximum,
+            sqrtPriceLimitX96: 0
+        });
+
+        uniswapV3Router.exactOutputSingle{value: msg.value}(params);
```

Comment:
I don't think `exactOutputSingle` works in this way because we pass `tokenIn` as WETH, not ETH
We may have to wrap ETH with WETH first using
```solidity
IWETH(WETH).deposit{value: msg.value}();
```
then do the same thing as `convertAndMint` (It would be possible to refactor the duplicated converting code as `_convert()` function)